File: MovieCatalogQw/app/src/androidTest/java/com/unlistedi/moviecatalogqw/MoviesFragmentTest.java
package com.unlistedi.moviecatalogqw;

import androidx.test.rule.ActivityTestRule;

import com.unlistedi.moviecatalogqw.testing.SingleFragmentActivity;
import com.unlistedi.moviecatalogqw.ui.movies.MoviesFragment;
import com.unlistedi.moviecatalogqw.utils.RecyclerViewItemCountAssertion;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;

public class MoviesFragmentTest {

    @Rule
    public ActivityTestRule<SingleFragmentActivity> activityRule = new ActivityTestRule<>(SingleFragmentActivity.class);
    private MoviesFragment moviesFragment = new MoviesFragment();

    @Before
    public void setUp() {
        activityRule.getActivity().setFragment(moviesFragment);
    }

    @Test
    public void loadMovies() {
        onView(withId(R.id.rvMovies)).check(matches(isDisplayed()));
        onView(withId(R.id.rvMovies)).check(new RecyclerViewItemCountAssertion(19));
    }
}

File: MovieCatalogueKu/app/src/androidTest/java/com/unlistedi/moviecatalogueku/ui/movies/MoviesFragmentTest.kt
package com.unlistedi.moviecatalogueku.ui.movies

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.rule.ActivityTestRule
import com.unlistedi.moviecatalogueku.R
import com.unlistedi.moviecatalogueku.testing.SingleFragmentActivity
import com.unlistedi.moviecatalogueku.utils.RecyclerViewItemCountAssertion
import org.junit.Before
import org.junit.Rule
import org.junit.Test

class MoviesFragmentTest {
    @get:Rule
    val activityRule = ActivityTestRule<SingleFragmentActivity>(SingleFragmentActivity::class.java)
    val moviesFragment = MoviesFragment()

    @Before
    fun setUp(){
        activityRule.activity.setFragment(moviesFragment)
    }

    @Test
    fun loadMovies(){
        onView(withId(R.id.rvMovies)).check(matches(isDisplayed()))
        onView(withId(R.id.rvMovies)).check(RecyclerViewItemCountAssertion(19))
    }
}